.gitignore

```diff
@@ -1,3 +1,4 @@
 vendor
 .idea
 /.phpunit.result.cache
+composer.lock
```

Comment:
Adding the composer.lock file to VCS is recommended for personal projects. Adding it to a library however will require to have updated version of the lockfile everytime you make a change to the composer.json file.

Also if there is a "security" or bugfix on a dependency and a new PR is sent without updating the composer.lock, there wont be any tests executed against the newer version of the dependencies.

Also see: 
https://getcomposer.org/doc/02-libraries.md#lock-file

Reply:
@mmachatschek 

> For your library you may commit the composer.lock file if you want to. This can help your team to always test against the same dependency versions. However, this lock file will not have any effect on other projects that depend on it. It only has an effect on the main project.